datafusion/physical-expr/src/window/nth_value.rs

```diff
@@ -210,9 +225,31 @@ impl PartitionEvaluator for NthValueEvaluator {
                 // We produce None if the window is empty.
                 return ScalarValue::try_from(arr.data_type());
             }
+
+            let mut valid_indices = Vec::new();
+            if self.ignore_nulls {
+                valid_indices = arr.nulls().unwrap().valid_indices().collect::<Vec<_>>();
+                if valid_indices.is_empty() {
+                    return ScalarValue::try_from(arr.data_type());
+                }
+            }
             match self.state.kind {
-                NthValueKind::First => ScalarValue::try_from_array(arr, range.start),
-                NthValueKind::Last => ScalarValue::try_from_array(arr, range.end - 1),
+                NthValueKind::First => {
+                    let index: usize = if self.ignore_nulls {
+                        valid_indices[0]
```

Comment:
As @mustafasrepo said, you will pick the first non-null value in the array, not the first non-null value in the range of the array.